packages/core/src/validate/run.ts

```diff
@@ -93,6 +123,9 @@ export function validate(solcOutput: SolcOutput, decodeSrc: SrcDecoder): Validat
         ];
 
         validation[contractDef.name].layout = extractStorageLayout(contractDef, decodeSrc, deref);
+        validation[contractDef.name].methods = [...findAll('FunctionDefinition', contractDef)]
+          .map(fnDef => getFunctionSignature(fnDef, deref))
+          .filter(Boolean) as string[];
```

Comment:
I think this would be a more explicit way to implement this.

```suggestion
          .filter(fnDef => ['external', 'public'].includes(fnDef.visibility))
          .map(fnDef => getFunctionSignature(fnDef, deref));
```

`getFunctionSignature` should always return a value, not only for external/public functions.

Reply:
right. I use to have a getFunctionSelector, but now that we use signature your version is better